addons/account/models/company.py

```diff
@@ -46,6 +60,71 @@ class ResCompany(models.Model):
 Best Regards,''')
 
     @api.multi
+    def _update_fields_values(self, fields):
+        """ Returns dict of write() values for synchronizing ``fields`` """
+        values = {}
+        for fname in fields:
+            field = self._fields[fname]
+            if field.type == 'many2one':
+                values[fname] = self[fname].id
+            elif field.type == 'one2many':
+                raise AssertionError(_('One2Many fields cannot be synchronized as part of `commercial_fields` or `address fields`'))
+            elif field.type == 'many2many':
+                values[fname] = [(6, 0, self[fname].ids)]
+            else:
+                values[fname] = self[fname]
+        return values
+
+    @api.model
+    def _accounting_company_fields(self):
+        """ Returns the list of fields that are managed by the accounting company
+        to which this company belongs. These fields are meant to be hidden on
+        companies that aren't `accounting companies` themselves, and will be
+        delegated to the parent `accounting companies`. The list is meant to be
+        eventually extended by inheriting classes. """
+        return ['currency_id', 'transfer_account_id']
+
+    def _accounting_sync_from_accounting_company(self):
+        """ Handle sync of accounting company fields when a new accounting company is set,
+        as if they were related fields """
+        accounting_company = self.accounting_company_id
+        if accounting_company != self:
+            sync_vals = accounting_company._update_fields_values(self._accounting_company_fields())
+            self.write(sync_vals)
+
+    def _accounting_sync_to_accounting_group(self):
+        """ Handle sync of accounting company fields to descendants """
+        accounting_company = self.accounting_company_id
+        if accounting_company == self:
+            sync_vals = accounting_company._update_fields_values(self._accounting_company_fields())
+            sync_group = self._get_accounting_subsidies().filtered(lambda r: r != self)  # All of same accounting group, without self
+            sync_group.write(sync_vals)
+
+    @api.multi
+    def _fields_sync(self, values):
+        self.ensure_one()
+        if values.get('parent_id'):
+            self._accounting_sync_from_accounting_company()
+        if self.child_ids:
+            self._accounting_sync_to_accounting_group()
```

Comment:
Above mechanism basically taken from `res.partner`. Some generic ORM solution for parent to child sync, would be interesting to me... @rco-odoo